lib/src/HttpUtils.cc

```diff
@@ -15,12 +15,217 @@
 #include "HttpUtils.h"
 #include <drogon/utils/Utilities.h>
 #include <trantor/utils/Logger.h>
+#include <map>
 #include <unordered_map>
+#include <mutex>
 
 namespace drogon
 {
 static std::unordered_map<std::string, std::string> customMime;
 
+// https://en.wikipedia.org/wiki/List_of_file_formats
+// https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types
+// https://www.digipres.org/formats/mime-types/
+// https://www.iana.org/assignments/media-types/media-types.xhtml
+// content type -> list of corresponding mime types, the first being the default
+// (more standard) one + the mime type to return in contentTypeToMime() when not
+// empty (mainly to return the charset for text types)
+static const std::map<ContentType,
+                      std::pair<std::list<std::string_view>, std::string_view>>
```

Comment:
I think it's better to use std::vector container.

Reply:
Or `std::unordered_map`?  Assuming there are no duplicates for `ContentType`, `std::unordered_map` can look up any elements in O(1) amortized time complexity, whereas `std::map` needs O(logN).

Reply:
Since ContentType is enum, `std::vector` works pretty better in terms of space complexity than `std::unordered_map` for `mimeTypeDatabase_`.  But for `fileTypeDatabase_`, I think it'd be better to use `std::unordered_map`.  Or, we can consider using a perfect hash function since we know all keys before compile time.  We can benefit from gperf or [frozen](https://github.com/serge-sans-paille/frozen).  gperf needs to generate the header file before compiling, so frozen might work better, but I'm sure it increases the compilation time (hopefully, it's negligible).  @an-tao What do you think?

Reply:
> Or `std::unordered_map`? Assuming there are no duplicates for `ContentType`, `std::unordered_map` can look up any elements in O(1) amortized time complexity, whereas `std::map` needs O(logN).


Because enumeration types can be converted to continuous int types, using std::vector is the fastest solution for lookup speed. However, in the source code, there will be a loss of direct correspondence between key and value, which can reduce code maintainability. Therefore, using unordered_map is a reasonable choice. Additionally, replacing std::list<string_view> with std::vector<string_view> is a better option.

Reply:
> hopefully, it's negligible

I think unordered_map is good enough, unless we have tested and confirmed that replacing the numerous immutable static maps in Drogon with the frozen library significantly improves runtime speed (such as achieving a performance improvement of over 10% in TFB tests), there is no need to introduce additional dependencies.

Reply:
> Because enumeration types can be converted to continuous int types, using std::vector is the fastest solution for lookup speed. However, in the source code, there will be a loss of direct correspondence between key and value, which can reduce code maintainability. Therefore, using unordered_map is a reasonable choice. Additionally, replacing std::list<string_view> with std::vector<string_view> is a better option.

I feel the benefit of time and space complexity of std::vector outweighs the maintainability as long as we leave a good comment on the vector.

> I think unordered_map is good enough, unless we have tested and confirmed that replacing the numerous immutable static maps in Drogon with the frozen library significantly improves runtime speed (such as achieving a performance improvement of over 10% in TFB tests), there is no need to introduce additional dependencies.

That sounds good to me.  For now, we can just leave room for further improvements and prioritize https://github.com/drogonframework/drogon/issues/1927 since this involves performance improvements as well.